src/openrct2/management/NewsItem.h

```diff
@@ -52,13 +52,75 @@ struct NewsItem
     uint16_t MonthYear;
     uint8_t Day;
     utf8 Text[256];
+
+    constexpr bool IsEmpty() const noexcept
+    {
+        return Type == NEWS_ITEM_NULL;
+    }
 };
 
-#define MAX_NEWS_ITEMS 61
+constexpr int32_t MAX_RECENT_NEWS_ITEMS = 11;
+constexpr int32_t MAX_OLD_NEWS_ITEMS = 50;
+constexpr int32_t MAX_NEWS_ITEMS = MAX_RECENT_NEWS_ITEMS + MAX_OLD_NEWS_ITEMS;
 
 extern const uint8_t news_type_properties[10];
 
-extern NewsItem gNewsItems[MAX_NEWS_ITEMS];
+struct NewsItemQueue
+{
+    NewsItem& operator[](size_t index);
+    NewsItem* At(int32_t index);
+    bool IsEmpty() const;
+    void Init();
+    uint16_t IncrementTicks();
+    NewsItem& Current()
+    {
+        return Recent[0];
+    }
+    const NewsItem& Current() const
+    {
+        return Recent[0];
+    }
+    NewsItem& Oldest()
+    {
+        return Old[0];
+    }
+    const NewsItem& Oldest() const
+    {
+        return Old[0];
```

Comment:
I'd prefer to move them to cpp for sake of compile times.

Reply:
Will do.
The reason I put them here is to allow inlining, and since these are straightforward I don't think they'd get in the way of compilation.

Reply:
Inlining… member functions? I doubt this would be done at compilation stage, but have no evidence to back it up. LTO on the other hand… that's not affected by the code being placed in the include or source file, though.

Reply:
It seems I'm in the wrong about that: https://isocpp.org/wiki/faq/inline-functions#inline-member-fns-more